workshops/geometric_pattern/README.md

```diff
@@ -0,0 +1,538 @@
+
+```js
+var NUM_CIRCLES = 12;
+var CIRCLE_DIA;
+var CIRCLE_RAD;
+var rVal;
+var gVal;
+var bVal;
+```
+
+Next, we'll set their initial values at the top of the `draw()` function:
+
+```js
+function draw() {
+  rVal = 255;
+  gVal = 0;
+  bVal = 0;
+
+...
+
+}
+```
+
+And then increment the values at the bottom of the outer for-loop in `draw()`, by adding these lines.
+
+```js
+...
+
+    rVal = rVal - 1;
+    gVal = gVal + 5;
+    bVal = bVal + 2;
+  }
+}
+```
+
+Here, we're decrementing the R value by 1, incrementing the G value by 5, and the B value by 2.
+
+You can also try adding those three lines within the inner for-loop, which will modify the colors within each row.
+
+Finally, we'll replace the arguments in `color()` with these variables, in both the `fill()` and `stroke()` function calls:
+
+```js
+fill(color(rVal,gVal,bVal));
+stroke(color(rVal,gVal,bVal));
+```
+
+Now, save and see the gradient effect you've applied throughout the pattern! Yay!
+
+## Part V: Publishing and Sharing
+
+### Downloading Your Masterpiece (OPTIONAL SECTION IDK HOW COOL THIS WILL BE CUZ IT'S JUST GONNA BE THE SAME THING SO WHO CARES)
+
+You can actually download the this cool pattern to your computer, to use as a desktop background or what have you. p5.js provides a function to download the canvas. (FACT CHECK THIS it's avail in processing but ive never done it in p5)
+
+### Making It Live
+
+Make sure all of your files are saved. Then, head on over to the terminal in C9 by pressing `alt+t` and type the following commands (pressing Enter after each one):
+
+- git add --all
+- git commit -m "Geometric Pattern Workshop"
+- git push
+
+Then, enter your GitHub username and password (careful here, as password will not be displayed).
+
+Congratulations! Your pattern is now live on `USERNAME.github.io/geometric_pattern` (replace `USERNAME` with your own GitHub username!)
+
+## Part VI: Animate It
+
+We can create a cool scrolling color effect by manipulating our colors to cycle. Right now we are resetting our initial RGB values each time `draw()` runs. But if we didn't reset every time, we could have rotating colors.
+
+```js
+function setup() {
+  createCanvas(480,600);
+  CIRCLE_DIA = width/NUM_CIRCLES;
+  CIRCLE_RAD = CIRCLE_DIA/2;
+  rVal = 255;
+  gVal = 0;
+  bVal = 0;
+}
+
+function draw() {
+  var alternatingRow = false;
+  var startX;
+
+...
+```
+
+If we move the three lines that set the initial RGB values to `setup()`, then we will be increment the values indefinitely. If we save and refresh our Live Preview, we'll see that everything is white. What's the deal?
+
+Since the max values of RGB are 255, 255, 255 (which makes white), any values above that will be white. The `draw()` function executes so many times per second that you can't see the color change progression.
+
+We can have color progression while keeping the values below 255 by modding each of these by 255 with the `%` operator. Let's change our incrementation code to incorporate this:
```

Comment:
True dat. I might take out this part?? Thoughts? Move it to like, the advanced demo or something. This part is too much explanation. Code is literally just moving stuff to `setup()` though. ALSO: In regards to your previous "why is this in `draw()` and not `setup()`" if we add this section, maybe we should write the whole workshop in `setup()`? and in this section we'd move the drawing code to `draw()`? or is that too much mental work in one workshop?